src/UrlHandlers/UrlHandler.php

```diff
@@ -365,6 +369,28 @@ protected function getMatchingUrlFragment(Request $request, $currentUrlFragment
         return false;
     }
 
+    /**
+     * Opportunity to add additional logic before getMatchingUrlFragment is executed
+     *
+     * @param Request $request
+     * @param string $currentUrlFragment
+     */
+    protected function onBeforeUrlFragmentMatched(Request $request, $currentUrlFragment = "")
```

Comment:
Hmm, the naming here is very dependant on the other function names which means it's not clear to others what these hooks would be for or when they might be called. It also implies that there is a positive match but in fact these are called either way.

Would it be better to call it onMatched() and only call it if $this->matchingUrl isn't false?

Reply:
Changed :)